src/script/interpreter.cpp

```diff
@@ -1279,18 +1279,28 @@ uint256 GetOutputsHash(const T& txTo)
 } // namespace
 
 template <class T>
-PrecomputedTransactionData::PrecomputedTransactionData(const T& txTo)
+void PrecomputedTransactionData::Init(const T& txTo)
```

Comment:
This interface allows for (mistakenly) calling `Init` on the same object with different inputs. i.e., once `ready` is set, any further calls to `Init` would be a no-op (at least in this commit). So it leaves open the potential for misuse. Could we design the interface in such away that makes this impossible?

Reply:
Maybe assert fail when you're trying to re-Init?

Reply:
Done